packages/core/src/browser/frontend-application.ts

```diff
@@ -55,6 +62,23 @@ export interface FrontendApplicationContribution {
      */
     initializeLayout?(app: FrontendApplication): MaybePromise<void>;
 }
+export interface OnWillStopEvent {
```

Comment:
i don't think we need it:
- `preventStop` can be replaced by boolean return value
- `onStay` by `setTimeout` on the contribution side